src/ui/abstract-option-select-ui-handler.ts

```diff
@@ -0,0 +1,465 @@
+import { globalScene } from "#app/global-scene";
+import { settings } from "#app/system/settings/settings-manager";
+import type { OptionSelectItem, OptionSelectModeConfig } from "#app/ui/interfaces/option-select-config";
+import MessageUiHandler from "#app/ui/message-ui-handler";
+import { ScrollBar } from "#app/ui/scroll-bar";
+import { TextStyle, addBBCodeTextObject, getBBCodeFrag, getTextStyleOptions } from "#app/ui/text";
+import { Mode } from "#app/ui/ui";
+import { addWindow } from "#app/ui/ui-theme";
+import { fixedNumber, isNullOrUndefined } from "#app/utils";
+import { Button } from "#enums/buttons";
+import type BBCodeText from "phaser3-rex-plugins/plugins/gameobjects/tagtext/bbcodetext/BBCodeText";
+import type { UIOptionSelectItem } from "./interfaces/option-select-ui-item";
+
+const SCROLLBAR_PADDING = 5;
+const SCROLLBAR_WIDTH = 3;
+const WINDOW_PADDING = 23;
+const DEFAULT_MAX_OPTIONS = 10;
+const NUM_PRE_COMPUTED_OPTIONS = 15;
+const DEFAULT_TEXT_STYLE = TextStyle.WINDOW;
+
+/**
+ * Generic handler for a menu with several options to choose from with a cursor.
+ * Given the proper {@linkcode OptionSelectModeConfig} the handler takes care of:
+ *  - Measure the size of all elements in the menu, including the size of any icon.
+ *  - Apply the required BBCode if an elements asks for a specific color.
+ *  - Creating the window of the appropriate size to hold all elements.
+ *  - Handle scrolling through the menu items, resizing as needed.
+ *  - Handle selecting the menu items, or cancelling out of the menu.
+ *
+ * At initialization the size of the first {@linkcode NUM_PRE_COMPUTED_OPTIONS} is measured.
+ * Then the window's size is updated as needed when a non initialized option needs to be displayed.
+ *
+ * @template T the specifc type of {@linkcode OptionSelectItem} that this handler displays
+ */
+export default abstract class AsbtractOptionSelectUiHandler<T extends OptionSelectItem> extends MessageUiHandler {
```

Comment:
Typo
```suggestion
export default abstract class AbstractOptionSelectUiHandler<T extends OptionSelectItem> extends MessageUiHandler {
```
This might affect other files too

Reply:
omg thank you... That handler is cursed, before it was the filename which had a typo